js/src/vm/PortableBaselineInterpret.cpp

```diff
@@ -3325,2718 +6031,2980 @@ PBIResult PortableBaselineInterpret(JSContext* cx_, State& state, Stack& stack,
   }
 #endif
 
+#ifdef ENABLE_COMPUTED_GOTO_DISPATCH
     goto* addresses[*pc];
+#else
+    (void)addresses;  // Avoid unused-local error. We keep the table
+                      // itself to avoid warnings (see note in IC
+                      // interpreter above).
+    switch (JSOp(*pc))
+#endif
+    {
+      CASE(Nop) { END_OP(Nop); }
+      CASE(NopIsAssignOp) { END_OP(NopIsAssignOp); }
+      CASE(Undefined) {
+        VIRTPUSH(StackVal(UndefinedValue()));
+        END_OP(Undefined);
+      }
+      CASE(Null) {
+        VIRTPUSH(StackVal(NullValue()));
+        END_OP(Null);
+      }
+      CASE(False) {
+        VIRTPUSH(StackVal(BooleanValue(false)));
+        END_OP(False);
+      }
+      CASE(True) {
+        VIRTPUSH(StackVal(BooleanValue(true)));
+        END_OP(True);
+      }
+      CASE(Int32) {
+        VIRTPUSH(StackVal(Int32Value(GET_INT32(pc))));
+        END_OP(Int32);
+      }
+      CASE(Zero) {
+        VIRTPUSH(StackVal(Int32Value(0)));
+        END_OP(Zero);
+      }
+      CASE(One) {
+        VIRTPUSH(StackVal(Int32Value(1)));
+        END_OP(One);
+      }
+      CASE(Int8) {
+        VIRTPUSH(StackVal(Int32Value(GET_INT8(pc))));
+        END_OP(Int8);
+      }
+      CASE(Uint16) {
+        VIRTPUSH(StackVal(Int32Value(GET_UINT16(pc))));
+        END_OP(Uint16);
+      }
+      CASE(Uint24) {
+        VIRTPUSH(StackVal(Int32Value(GET_UINT24(pc))));
+        END_OP(Uint24);
+      }
+      CASE(Double) {
+        VIRTPUSH(StackVal(GET_INLINE_VALUE(pc)));
+        END_OP(Double);
+      }
+      CASE(BigInt) {
+        VIRTPUSH(StackVal(JS::BigIntValue(frame->script()->getBigInt(pc))));
+        END_OP(BigInt);
+      }
+      CASE(String) {
+        VIRTPUSH(StackVal(StringValue(frame->script()->getString(pc))));
+        END_OP(String);
+      }
+      CASE(Symbol) {
+        VIRTPUSH(StackVal(SymbolValue(
+            ctx.frameMgr.cxForLocalUseOnly()->wellKnownSymbols().get(
```

Comment:
question: what happens in the case when it is not a well-known symbol?

Reply:
The bytecode emitter only uses the opcode in cases where it is; `get` is a [checked index-accessor](https://searchfox.org/mozilla-central/rev/91f6127b6f591da4037821791c345147d9a575da/js/src/vm/Runtime.h#153) so will be infallible unless the emitter uses a bad enum index.